virtualizarr/tests/test_integration.py

```diff
@@ -230,24 +221,24 @@ def test_non_dimension_coordinates(self, tmpdir, roundtrip_func, hdf_backend):
         for coord in ds.coords:
             assert ds.coords[coord].attrs == roundtrip.coords[coord].attrs
 
-    def test_datetime64_dtype_fill_value(self, tmpdir, roundtrip_func):
+    # In Zarr V3 we cannot use a data type of dat
+    def test_datetime64_dtype_fill_value(
+        self, tmpdir, roundtrip_func, array_v3_metadata
+    ):
         chunks_dict = {
             "0.0.0": {"path": "/foo.nc", "offset": 100, "length": 100},
         }
         manifest = ChunkManifest(entries=chunks_dict)
         chunks = (1, 1, 1)
         shape = (1, 1, 1)
-        zarray = ZArray(
-            chunks=chunks,
-            compressor={"id": "zlib", "level": 1},
-            dtype=np.dtype("<M8[ns]"),
-            # fill_value=0.0,
-            filters=None,
-            order="C",
+        metadata = array_v3_metadata(
             shape=shape,
-            zarr_format=2,
+            chunks=chunks,
+            codecs=[{"name": "numcodecs.zlib", "configuration": {"level": 1}}],
+            # Cannot do in Zarr v3
+            data_type=np.dtype("M8[ns]"),
```

Comment:
This throws an error `KeyError: '<M8[ns]'` from https://github.com/zarr-developers/zarr-python/blob/v3.0.2/src/zarr/core/metadata/v3.py#L704. Datetime dtypes are not yet supported in zarr-python 3 (see https://zarr.readthedocs.io/en/stable/user-guide/v3_migration.html:

>The following features that were supported by Zarr-Python 2 have not been ported to Zarr-Python 3 yet:
>- Datetime and timedelta dtypes ([#2616](https://github.com/zarr-developers/zarr-python/issues/2616))

Reply:
This restriction must apply to loadable variables too? In which case this is potentially a blocker?

Reply:
@TomNicholas I might be misunderstanding here but `loadable_variables` will never reach the `ArrayV3Metadata`.  In the interim until there is upstream support for Datetime `dtypes` should our readers introspect variables and force Datetime variables as `loadable_variables`?  I know this seems a bit heavy handed but it might give users a better experience in the short term.

Reply:
> `loadable_variables` will never reach the ArrayV3Metadata

No, but we're still saving the `loadable_variables` into zarr v3 via icechunk as native chunks eventually. So if zarr doesn't support this dtype don't we still have a problem? Or is there some encoding step that dodges the issue?

> force Datetime variables as loadable_variables?

I would be fine with that, and actually after #353 that would mostly happen by default for realistic datasets.

Reply:
@TomNicholas As discussed in https://github.com/earth-mover/icechunk/issues/119 if we are using `xarray` CF datetime encoding the raw array dtype for serialization will be `int` and `xarray` will use the `units` `attr` for decoding.

Reply:
Oh great, thanks for the explanation/reminder @sharkinsspatial .

Reply:
Should we include any test or documentation about the xarray method of datetime encoding @sharkinsspatial @TomNicholas ? Perhaps as a separate issue since it's not directly related to this PR, apart from being an approach to get around the current zarr-python datetime/timedelta data type limitation.

Reply:
Raised #437 to track that.